doc/SONiC_YANG_Model_Guidelines.md

```diff
@@ -71,8 +178,85 @@ Leaf names are same PACKET_ACTION, IP_TYPE and PRIORITY, which are .
             }
 ```
 
+### 9. Use appropriate data type for each leaf. Use enum, range and pattern as needed for defining data syntax constraint. 'leaf-list' is defined when array of values are used. 
+
+
+Example :
+#### ABNF
+
+```
+key           = ACL_TABLE:name          ; acl_table_name must be unique
+;field        = value
+policy_desc   = 1*255VCHAR              ; name of the ACL policy table description
+type          = "MIRROR"/"L2"/"L3"/"L3v6"    ; type of acl table, every type of
+                                        ; table defines the match/action a
+                                        ; specific set of match and actions.
+stage 		  = "INGRESS"/"EGRESS" 	    ; Stage at which policy is applied
+ports         = [0-max_ports]*port_name ; the ports to which this ACL
+                                        ; table is applied, can be emtry
+                                        ; value annotations
+port_name     = 1*64VCHAR               ; name of the port, must be unique
+max_ports     = 1*5DIGIT                ; number of ports supported on the chip
+```
 
-### 3. Data Node/Object Hierarchy of the an objects in YANG models will be same as for all the fields at same hierarchy in Config DB. If any exception is created then it must be recorded properly with comment under object level in YANG models. To see an example of a comment, please refer Guideline # 7.
+#### YANG
+```
+container ACL_TABLE {
+	list ACL_TABLE_LIST {
+		key aclname;
+			
+		leaf aclname {
+			type string;
+		}
+
+		leaf policy_desc {
+			type string {
+				length 1..255 {
+					error-app-tag policy-desc-invalid-length;
+				}
+			}
+		}
+
+		leaf stage {
+			type enumeration {
+				enum INGRESS;
+				enum EGRESS;
+			}
+		}
+
+		leaf type {
+			type enumeration {
+				enum MIRROR;
+				enum L2;
+				enum L3;
+				enum L3V6;
+			}
+		}
+
+		leaf-list ports {
+			type leafref {
+				path "/prt:sonic-port/prt:PORT/prt:ifname";
+			}
+		}
+	}
+}
+```
+
+### 10.  Use IETF data types for leaf type first if applicable (RFC 6021) . Declare new type (say SONiC types) only if IETF type is not applicable. All SONiC Types must be part of same header type or common YANG model.
+Example:
+
+#### YANG
+```
+                    leaf SRC_IP {
+                        type inet:ipv4-prefix; <<<<
+                    }
+
+                    leaf DST_IP {
+                        type inet:ipv4-prefix;
+                    }
+```
+
+### 11. Data Node/Object Hierarchy of the an objects in YANG models will be same as for all the fields at same hierarchy in Config DB. If any exception is created then it must be recorded properly with comment under object level in YANG models. To see an example of a comment, please refer Guideline #20.
```

Comment:
Let us not hard code the number, as soon it may go out of sync. Instead, something like "please refer the guidelines on how to comment'

Reply:
I agree.